### PR TITLE
Document installation with `cargo-binstall`

### PR DESCRIPTION
**Pull request by ilyagr**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by ilyagr** [comment]
One alternative I'm considering is we could introduce a "Cross-Platform" section after the "Windows" section and move `cargo-binstall` there. I think I'm mildly leaning to what I have now, but let me know what you prefer.

**Review by martinvonz — APPROVED** [review]
I think what you have now seems good. Thanks.

File: docs/install-and-setup.md [hunk 1]
```diff
@@ -113,6 +107,26 @@ or:
 cargo install --locked --bin jj jj-cli
 ```
 
+
+#### Homebrew
```
Comment (thoughtpolice) [inline docs/install-and-setup.md:111]: I honestly wonder if we should consolidate the Homebrew sections for Linux/Mac. Not a big deal I guess.

**Review by thoughtpolice — COMMENTED** [review]
I think this is OK, I'm personally neutral on supporting `cargo binstall`; it's mostly a convenience. That said I think the installation page just needs an overhaul. Maybe I'll give that a shot later.3

File: docs/install-and-setup.md (hunk 1 above)
Comment (ilyagr) [reply docs/install-and-setup.md:111]: If only homebrew supported Windows, I could put it and binstall in the same section...

**Comment by ilyagr** [comment]
BTW, while I don't use BInstall for jj, I use it constantly for development tools like nextest and insta, as well as tools that happen to be written in rust like broot (which I don't use much, but that's another story) or term-transcript

**Pull request merged**